### PR TITLE
Add Scikit-HEP & zfit requests to environment.yml

### PR DESCRIPTION
**Pull request by eduardo-rodrigues**

### CODE REVIEW

**Review by vuillaut — APPROVED** [review]

**Comment by eduardo-rodrigues** [comment]
Thanks, well spotted! I can't believe I did not see that conda-forge was not present :D.

**Comment by vuillaut** [comment]
> Thanks, well spotted! I can't believe I did not see that conda-forge was not present :D.

No problem, I saw the CI failing.
I let you merge if ready.

**Comment by eduardo-rodrigues** [comment]
OK, this "trivial" update is a bit more subtle. I want to use the latest zfit as well but that's not available in Conda because of required dependencies dragging a bit. I will revert to using pip here, and will add a little thing more. Next time I test in Binder or in a venv locally ...

**Comment by eduardo-rodrigues** [comment]
OK, @vuillaut, this is now ready. (I'm talking to the zfit author/friend) to get a latest and greatest release for Python 3.9 but that will come in a subsequent MR. At least this first one provides the base for Scikit-HEP stuff now rather than later :-).

**Comment by eduardo-rodrigues** [comment]
I can't approve and merge so please do if happy ("2 workflows awaiting approval. First-time contributors need a maintainer to approve running workflows)".

**Comment by vuillaut** [comment]
@eduardo-rodrigues it seems this introduces an incompatibility (see test failing), could you have a look pls?

**Comment by maxnoe** [comment]
erfa 2.0 was just released, seems that astropy still requires 1.7

**Comment by eduardo-rodrigues** [comment]
Weird - I see an error `ImportError: erfa version 1.7 or later must be installed to use Astropy. erfa is not installed.` but see version 1.7.3 installed, so should be fine. Will look further ...

On a related topic: it might be good to help the resolver by suggestion lower and/or upper limits to package versions as I did. Also, the YAML file tries to pin Python 3.9 but `python=3.9` misses an `=` ;-). And in any case I would not be that strict allowing at least 3.8. Speaking for myself I have a Windows (I know!) machine and 3.9 isn't available in anaconda, so will be either running 3.8 locally or 3.9 on Binder ... In short, I would allow at least 3.8 and 3.9.

In fact the tests run on 3.8 and, if supporting 3.9, they should also run on 3.9, if not for anything else to check (in)compatibilities.

**Comment by eduardo-rodrigues** [comment]
> Also, the YAML file tries to pin Python 3.9 but python=3.9 misses an = ;-).

I take that back. Got confused again by the syntax of requirement files and others.

And I see the tests actually running on Python 3.9.

**Comment by vuillaut** [comment]
`package zfit-0.5.2-py_0 requires python >=3.6,<3.8, but none of the providers can be installed`

**Comment by eduardo-rodrigues** [comment]
zfit and phasespace don't have conda releases since quite a few. I have been in touch with the author and that is why I got back to installing via pip, not conda. CC @mayou36 in case he wants to comment further.

**Comment by maxnoe** [comment]
@eduardo-rodrigues just trying to resolve the conflict that seems to arise here. Will put it back into pip.

**Comment by maxnoe** [comment]
> On a related topic: it might be good to help the resolver by suggestion lower and/or upper limits to package versions as I did. Also, the YAML file tries to pin Python 3.9 but python=3.9 misses an = ;-). 

conda environments use a single `=`, not `==` if you just want to pin the major version, not the minor.

**Comment by eduardo-rodrigues** [comment]
> @eduardo-rodrigues just trying to resolve the conflict that seems to arise here. Will put it back into pip.

OK, fair enough. I commented in case you did not realise the feature (I did not at first).

> Right now, _many_ large packages are installed via pip as depedencies of those two, which I wanted to avoid by installing them via conda. If we don't get conda packages for those in time, we can put the main depedencies of those in the conda section and then pip install zfit and phasespace

I'm trying to get conda installations where possible as well ...

Thanks.

File: environment.yml
Comment (mayou36) [inline environment.yml:36]:
```suggestion
```
remove as they are not up to date

**Comment by mayou36** [comment]
> Right now, _many_ large packages are installed via pip as depedencies of those two, which I wanted to avoid by installing them via conda. If we don't get conda packages for those in time, we can put the main depedencies of those in the conda section and then pip install zfit and phasespace

Just to confirm, the correct zfit and hepstats packages are underway.

The main dependency would be TensorFlow, which we can't (as it is behind), the others seem fine so far.

File: environment.yml
Comment (mayou36) [inline environment.yml:43]:
```suggestion
    - zfit>0.6,<0.8
```

Comment (mayou36) [reply environment.yml:43]: The new version is gonna be 0.7.x

File: environment.yml
Comment (maxnoe) [reply environment.yml:36]: This installs 2.4.2, do you really need the very latest 2.5 from pypi?

**Comment by vuillaut** [comment]
The install is now working.
I will wait for @eduardo-rodrigues reply to @mayou36 comment and fix zfit version.

Do you really need tensorflow?

File: environment.yml
Comment (eduardo-rodrigues) [reply environment.yml:43]: Thanks. I was hoping to do that change in a subsequent MR but this one ended up being far less trivial than I thought. No harm to do it now and here.

**Comment by mayou36** [comment]
> Do you really need tensorflow?

Yes, it is the computational backend, so as much as other libraries may need numpy

File: environment.yml
Comment (eduardo-rodrigues) [reply environment.yml:40]: That will only be true for the upcoming zfit release with Python 3.9 support, if I'm not mistaken. But @mayou36 can confirm.

Comment (maxnoe) [reply environment.yml:40]: If it helps we can go to 3.8 for the moment.

Comment (mayou36) [reply environment.yml:40]: Yes, this will be needed. Unfortunately (or also fortunately), tensorflow still has quite a lot of breaking changes in their API and therefore 2.5 is needed.

For the release, I think in 30 mins both zfit (already out) and hepstats should work with 3.9

Comment (maxnoe) [reply environment.yml:41]: I just got the report from an apple user, that 3.9 is required to run on M1. So we might have a problem here.

Do we really need zfit / phasespace in the environment? Or would a demo be enough for the school?

Comment (mayou36) [reply environment.yml:41]: I would say yes, but it should not be a problem, they all run in 3.9 (the CIs just passed). We simply need the hepstats pip release (~15 mins and we have it) and it should work

**Comment by mayou36** [comment]
So hepstats and zfit releases for Python 3.9 via pip are out, this shouldn't be a bottleneck anymore. Just let me know in case there is any problem with it

**Comment by eduardo-rodrigues** [comment]
> So hepstats and zfit releases for Python 3.9 via pip are out, this shouldn't be a bottleneck anymore. Just let me know in case there is any problem with it

Big thanks for this @mayou36 👍 ! (hepstats will even be available in conda but probably only tomorrow once the bot does its magic.)

**Comment by eduardo-rodrigues** [comment]
For info and completeness you added iminuit but that's coming in fact as part of scikit-hep, which is a metapackage with a good dozen org packages made readily available in one line. Dunno if you want to make this clean-up here or else I may do it with the odd addition I'm likely to have to make in the next few days (minor update compared to this one). Up to you.

Else we're done. Thank you all for your precious help!

**Comment by maxnoe** [comment]
Let's merge and do any updates in another PR?

**Pull request merged**
